src/rov/rov_led_controller/rov_led_controller/led_controller.py

```diff
@@ -1,129 +1,200 @@
             self.publisher_.publish(animations_msg)
             self.get_logger().info('Sent preprogrammed animations to UI')
-        elif msg.data == 'get_colors':
+        elif msgs[0] == 'get_colors':
             colors_msg = String()
             colors_msg.data = "\n".join(color for color in self.colors)
             self.publisher_.publish(colors_msg)
             self.get_logger().info('Sent preprogrammed colors to UI')
         else:
             # Check if the requested animation exists
-            if msg.data in self.animations:
+            if msgs[0] in self.animations:
                 # Call the method corresponding to the requested animation
-                
-                if ledState and (lastAnimation != msg.data):
-                    exec(f"self.{msg.data}({ledColor}, 0.1)")()
-                else:
-                    PIXELS.brightness(0)
-                lastAnimation = msg.data
-            elif msg.data in self.colors:
-                lastColor = ledColor
-                if ledState and (lastColor != msg.data):
-                    exec(f"self.{lastAnimation}({msg.data}, 0.1)")()
-
-
-    def on_set_rov_gpio(self, message: PinState):
-        if message.pin not in LED_PIN:
-            pass
-        ledState = message.state
+                self.currentAnimation = msgs[0]
+                self.get_logger().info('New animation received: %s' % msgs[0])
+                self.animations[msgs[0]](self.colors[self.ledColor])
+            elif msgs[0] in self.colors:
+                # Change the color if message changes colors
+                self.ledColor = msgs[0]
+                self.get_logger().info('New color received: %s' % msgs[0])
+                self.animations[self.currentAnimation](self.colors[msgs[0]])
+            
+
     
-    def color_chase(color, wait=0.1):
+    def color_chase(self, color, wait=0.1):
+        """
+        Animates a single color across the LED strip
+        """
         for i in range(NUM_LIGHTS):
-            PIXELS[i] = color
+            PIXELS.RGBto3Bytes(i, color[0], color[1], color[2])
             time.sleep(wait)
-            PIXELS.show()
+        PIXELS.LED_show()
         time.sleep(0.5)
 
-    def rainbow_cycle(color, wait=0.1):
-        for j in range(255):
-            for i in range(NUM_LIGHTS):
-                rc_index = (i * 256 // NUM_LIGHTS) + j
-                PIXELS[i] = colorwheel(rc_index & 255)
-            PIXELS.show()
-            time.sleep(wait)
 
-    def pulse(color, wait=0.1):
-        PIXELS.fill(color)
+    def rainbow_cycle(self, color, wait=0.1):
+        """
+        Cycles a wave of rainbow over the LEDs
+        """
+        RGBincrement = float(255 * 3.0) / NUM_LIGHTS
+        # LED RGB values
+        tempR = self.rainbowR
+        tempG = self.rainbowG
+        tempB = self.rainbowB
+
+        # Assigning LED RGB values
+        for i in range(NUM_LIGHTS):
```

Comment:
You couuuuld use HSV and cycle through Hues if you so chose.. colorsys has a thing for that and I think it's in the standard library (so no extra dependency)